src/apps/service/apps/combined/util.clj

```diff
@@ -22,6 +22,12 @@
         (apply-offset params)
         (apply-limit params))))
 
+(defn combine-concurrent-app-listings
```

Comment:
I am, of course, late to the party, but I suppose it's maybe still worth noting: I don't know if it's worth it, but I'd have probably rolled this into `combine-app-listings` something like:

```clojure
(let [maybe-deref (fn [r] (if (future? r) (deref r) r))
       results (mapv maybe-deref results)
...etc
```

not sure if that's better or worse, but I do somewhat like a model where we expect things that take deref-able arguments to also accept already-deref'd values in the same function, such that it's mostly transparent.

(just food for thought, anyway!)

Reply:
I like that idea. I'll add it in the next PR.